lib/types/duration.js

```diff
@@ -28,173 +30,179 @@ const iso8601WeekRegex = /P(\d+)W/;
 const iso8601AlternateRegex =
   /P(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})/;
 
-/**
- * Creates a new instance of {@link Duration}.
- * @classdesc
- * Represents a duration. A duration stores separately months, days, and seconds due to the fact that the number of
- * days in a month varies, and a day can have 23 or 25 hours if a daylight saving is involved.
- * @param {Number} months The number of months.
- * @param {Number} days The number of days.
- * @param {Number|Long} nanoseconds The number of nanoseconds.
- * @constructor
- */
-function Duration(months, days, nanoseconds) {
+class Duration {
   /**
-   * Gets the number of months.
-   * @type {Number}
+   * Creates a new instance of {@link Duration}.
+   * @classdesc
+   * Represents a duration. A duration stores separately months, days, and seconds due to the fact that the number of
+   * days in a month varies, and a day can have 23 or 25 hours if a daylight saving is involved.
+   * @param {Number} months The number of months.
+   * @param {Number} days The number of days.
+   * @param {Number|Long} nanoseconds The number of nanoseconds.
+   * @constructor
    */
-  this.months = months;
-  /**
-   * Gets the number of days.
-   * @type {Number}
-   */
-  this.days = days;
+  constructor(months, days, nanoseconds) {
+    if (nanoseconds instanceof Long)
+      this.internal = rust.DurationWrapper.new(
+        months,
+        days,
+        longToBigint(nanoseconds),
+      );
```

Comment:
So this is an API-compatible modernisation of Datastax API as suggested by @Chlor87?

Reply:
Yes, it's cleaner way of declaring class which was suggested by Kuba.